### ISSUE TITLE
zpool upgrade should have an option to explicitly specify the compatibility level

### ISSUE THREAD

**Issue by siebenmann** [openzfs/zfs#12262]
### Describe the feature would like to see added to OpenZFS

In the current development version, `zpool upgrade` supports upgrading to only specified compatibility feature sets. However, currently these feature sets must be specified indirectly by setting a pool property (and this has an issue, #12261). It would be nice if you could directly specify a compatibility feature set in `zpool upgrade` for at least pools that don't have a `compatibility` property set.

The friendliest behavior if a pool has a `compatibility` set is that the specified compatibility features from the command line are treated as another compatibility restraint, and only features in the union of the two are enabled. This would be the same as specifying multiple feature sets in the `compatibility` property and so presumably would be easy to implement.

The obvious approach would be a `-o compatibility=....` command line argument, but that might not be the best, since `zpool upgrade` doesn't otherwise take pool properties or things like them.

### How will this feature improve OpenZFS?

There are two audiences for this change. First, there are cautious system administrators who want to be absolutely sure of what features they get with a `zpool upgrade`, without having to check and double-check all of their pools' `compatibility` settings. Second, people who want to only partially upgrade their pools for now without setting an explicit `compatibility` level on the pool.

**Comment by colmbuckley** [comment]
I agree that this would be a nice feature (and not, I think, too difficult). I'm afraid I doubt I will have any time in the immediate future to look into implementing it, but I will definitely keep it in mind for later in the summer.
